clang/include/clang/Driver/ToolChain.h

```diff
@@ -339,6 +339,12 @@ class ToolChain {
                      Action::OffloadKind DeviceOffloadKind,
                      SmallVectorImpl<llvm::opt::Arg *> *AllocatedArgs) const;
 
+  /// TranslateBackendTargetArgs - populate an argument list with the specific
+  /// options that are determined to be passed along to an offline backend
+  /// target compiler.
+  void TranslateBackendTargetArgs(const llvm::opt::ArgList &Args,
+                                  llvm::opt::ArgStringList &CmdArgs) const;
```

Comment:
Could you please clarify why porting this function to the base toolchain class is required? I'm sure there's rationale that I simply haven't gathered from the code by this stage.